helix-view/src/document.rs

```diff
@@ -506,6 +513,28 @@ impl Document {
         }
     }
 
+    /// Get size (in bytes) of currently opened file.
+    /// If no file is opened, or there is some issue with the file,
+    /// then size will be assumed as 0.
+    pub fn get_file_size(&self) -> u64 {
+        match &self.path {
+            Some(path) => {
+                let file = std::fs::File::open(path);
+                if file.is_err() {
+                    return 0;
+                }
+
+                let metadata = file.unwrap().metadata();
+                if metadata.is_err() {
+                    return 0;
+                }
+
+                metadata.unwrap().len()
+            }
+            None => 0,
+        }
```

Comment:
```suggestion
        self.path
            .as_ref()
            .and_then(|path| {
                let file = std::fs::File::open(path).ok()?;
                let size = file.metadata().ok()?.len();
                Some(size)
            })
            .unwrap_or_default()
```

This feels off being a method on Document since it returns the file size on disk rather than the size of the (potentially) unsaved file in the buffer. Might be better suited as utility function.